File: app/src/main/java/org/amoustakos/boilerplate/injection/module/DialogModule.java
package org.amoustakos.boilerplate.injection.module;

import android.app.Dialog;
import android.content.Context;

import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext;

import dagger.Module;
import dagger.Provides;


@Module
public class DialogModule {
    private Dialog mDialog;

    public DialogModule(Dialog dialog) {
        mDialog = dialog;
    }


    @Provides
    @ActivityContext
    Context providesContext() {
        return mDialog.getContext();
    }
}

File: app/src/main/java/org/amoustakos/boilerplate/injection/module/injectors/DialogModule.kt
package org.amoustakos.boilerplate.injection.module.injectors

import android.app.Dialog
import android.content.Context
import dagger.Module
import dagger.Provides
import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext


@Module
class DialogModule(
		private val mDialog: Dialog
) {

	@Provides
	@ActivityContext
	internal fun providesContext(): Context = mDialog.context

}
